File: src/test/java/com/enonic/xp/app/users/apis/graphql/UtilsTest.java
package com.enonic.xp.app.users.apis.graphql;

import org.junit.jupiter.api.Test;

import com.enonic.xp.testing.ScriptTestSupport;

public class UtilsTest
    extends ScriptTestSupport
{
    @Test
    public void toIntFromNumber()
    {
        runFunction( "/com/enonic/xp/app/users/apis/graphql/utils-test.js", "toIntFromNumber" );
    }

    @Test
    public void toIntFromString()
    {
        runFunction( "/com/enonic/xp/app/users/apis/graphql/utils-test.js", "toIntFromString" );
    }

    @Test
    public void toIntNullReturnsDefault()
    {
        runFunction( "/com/enonic/xp/app/users/apis/graphql/utils-test.js", "toIntNullReturnsDefault" );
    }
}

File: src/test/resources/com/enonic/xp/app/users/apis/graphql/utils-test.js
var t = require('/lib/xp/testing');
var utils = require('/apis/graphql/utils');

exports.toIntFromNumber = function () {
    t.assertEquals(42, utils.toInt(42));
};

exports.toIntFromString = function () {
    t.assertEquals(42, utils.toInt('42'));
};

exports.toIntNullReturnsDefault = function () {
    t.assertEquals(7, utils.toInt(null, 7));
};
